scene/gui/check_button.cpp

```diff
@@ -52,14 +71,30 @@ Size2 CheckButton::get_minimum_size() const {
 	Size2 tex_size = get_icon_size();
 	minsize.width += tex_size.width;
 	if (get_text().length() > 0) {
-		minsize.width += MAX(0, get_theme_constant(SNAME("h_separation")));
+		minsize.width += MAX(0, theme_cache.h_separation);
 	}
-	Ref<StyleBox> sb = get_theme_stylebox(SNAME("normal"));
-	minsize.height = MAX(minsize.height, tex_size.height + sb->get_margin(SIDE_TOP) + sb->get_margin(SIDE_BOTTOM));
+	minsize.height = MAX(minsize.height, tex_size.height + theme_cache.normal_style->get_margin(SIDE_TOP) + theme_cache.normal_style->get_margin(SIDE_BOTTOM));
 
 	return minsize;
 }
 
+void CheckButton::_update_theme_item_cache() {
+	Button::_update_theme_item_cache();
+
+	theme_cache.h_separation = get_theme_constant(SNAME("h_separation"));
+	theme_cache.check_v_adjust = get_theme_constant(SNAME("check_v_adjust"));
+	theme_cache.normal_style = get_theme_stylebox(SNAME("normal"));
+
+	theme_cache.checked = get_theme_icon(SNAME("on"));
+	theme_cache.unchecked = get_theme_icon(SNAME("off"));
+	theme_cache.checked_disabled = get_theme_icon(SNAME("on_disabled"));
+	theme_cache.unchecked_disabled = get_theme_icon(SNAME("off_disabled"));
+	theme_cache.checked_mirrored = get_theme_icon(SNAME("on_mirrored"));
+	theme_cache.unchecked_mirrored = get_theme_icon(SNAME("off_mirrored"));
+	theme_cache.checked_disabled_mirrored = get_theme_icon(SNAME("on_disabled_mirrored"));
+	theme_cache.unchecked_disabled_mirrored = get_theme_icon(SNAME("off_disabled_mirrored"));
```

Comment:
Oof those theme item names... I guess the names you're using for the theme cache would be good candidates for renaming the theme items (but I'm sure you have this planned).

Reply:
Yes, cache names are from CheckBox, which fared better 😄 Definitely need to harmonize those.